contributors/devel/sig-architecture/feature-gates.md

```diff
@@ -0,0 +1,61 @@
+# Feature gate progression in Kubernetes
+
+Feature flags are how Kubernetes tracks progression of functionality that is available when deployed.
+
+## Lifecycle
+
+Feature progress through `Alpha` -> `Beta` -> `GA`. Sometimes we end up deciding that a feature is not going to be supported and we end up marking them as `Deprecated`.
+
+When we add a feature flag, we basically add if/else conditions to ensure that a feature is ONLY activated when either the default is on or if the deployer has switched it on explicitly.
+
+## Fields in the specification are
+	// Default is the default enablement state for the feature. possible values are true/false.
+	Default bool
+	
+    // LockToDefault indicates that the feature is locked to its default and cannot be changed. possible values are true/false.
+	LockToDefault bool
+	
+    // PreRelease indicates the maturity level of the feature. possible values are "featuregate.Alpha", "featuregate.Beta", "featuregate.GA", "featuregate.Deprecated"
+	PreRelease prerelease
+
+
+## Alpha Features
+
+* `PreRelease` is set to `featuregate.Alpha`
+* `Default` is always set to `false`
+* `LockToDefault` is not set. Defaults to `false`
+
+This enables folks to switch on the feature using the command line. By default it is not switched on.
```

Comment:
Something like "All API changes must start with an alpha gate, which makes it possible to rollback from future versions" ?